tfchain_bridge/pkg/bridge/withdraw.go

```diff
@@ -0,0 +1,133 @@
+package bridge
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/substrate-client"
+	subpkg "github.com/threefoldtech/tfchain_bridge/pkg/substrate"
+)
+
+func (bridge *Bridge) handleWithdrawCreated(ctx context.Context, withdraw subpkg.WithdrawCreatedEvent) error {
+	burned, err := bridge.subClient.IsBurnedAlready(types.U64(withdraw.ID))
+	if err != nil {
+		return err
+	}
+
+	if burned {
+		log.Info().Uint64("ID", uint64(withdraw.ID)).Msgf("tx is burned already, skipping...")
+		return errors.New("tx burned already")
+	}
+
+	if err := bridge.wallet.CheckAccount(withdraw.Target); err != nil {
+		log.Info().Uint64("ID", uint64(withdraw.ID)).Msg("tx is an invalid burn transaction, minting on chain again...")
+		mintID := fmt.Sprintf("refund-%d", withdraw.ID)
+		err := bridge.handleMint(big.NewInt(int64(withdraw.Amount)), substrate.AccountID(withdraw.Source), mintID)
+		for err != nil {
+			log.Err(err).Msg("error while minting on chain again for invalid withdraw")
+
+			select {
+			case <-ctx.Done():
+				return err
+			case <-time.After(10 * time.Second):
+				err = bridge.handleMint(big.NewInt(int64(withdraw.Amount)), substrate.AccountID(withdraw.Source), mintID)
+			}
+		}
+		if err != nil {
+			return errors.Wrap(err, "failed to handle mint on chain again for invalid withdraw")
+		}
+
+		log.Info().Uint64("ID", uint64(withdraw.ID)).Msg("setting invalid burn transaction as executed")
+		err = bridge.subClient.SetBurnTransactionExecuted(bridge.subClient.Identity, withdraw.ID)
+		for err != nil {
+			log.Err(err).Msg("error while setting burn transaction as executed")
+
+			select {
+			case <-ctx.Done():
+				return err
+			case <-time.After(10 * time.Second):
+				err = bridge.subClient.SetBurnTransactionExecuted(bridge.subClient.Identity, withdraw.ID)
+			}
+		}
+		if err != nil {
+			return errors.Wrap(err, "failed to handle setting burn transaction as executed")
+		}
+
+		return nil
+	}
+
+	amount := big.NewInt(int64(withdraw.Amount))
+	signature, sequenceNumber, err := bridge.wallet.CreatePaymentAndReturnSignature(ctx, withdraw.Target, amount.Uint64(), withdraw.ID)
+	if err != nil {
+		return err
+	}
+	log.Info().Msgf("stellar account sequence number: %d", sequenceNumber)
+
+	return bridge.subClient.ProposeBurnTransactionOrAddSig(bridge.subClient.Identity, withdraw.ID, withdraw.Target, amount, signature, bridge.wallet.GetKeypair().Address(), sequenceNumber)
```

Comment:
The first argument of ProposeBurnTransactionOrAddSig is the Identity, but both  are part of subclient, therefore this method should get the identity from the struct and not have it as argument.